subprojects/gdk_rust/gdk_electrum/src/lib.rs

```diff
@@ -660,98 +544,55 @@ impl Session<Error> for ElectrumSession {
 
         let notify_blocks = self.notify.clone();
 
-        let mut last_tip = tipper.tip()?;
-        info!("tip is {:?}", last_tip);
-        notify_block(notify_blocks.clone(), last_tip);
-
         let (close_tipper, r) = channel();
         self.closer.senders.push(close_tipper);
+        let tipper_url = self.url.clone();
         let tipper_handle = thread::spawn(move || {
             info!("starting tipper thread");
             loop {
-                if wait_or_close(&r, sync_interval) {
-                    info!("closing tipper thread");
-                    break;
-                }
-
-                match tipper.tip() {
-                    Ok(current_tip) => {
-                        if last_tip != current_tip {
-                            last_tip = current_tip;
-                            info!("tip is {:?}", last_tip);
-                            notify_block(notify_blocks.clone(), last_tip);
-                        }
-                    }
-                    Err(e) => {
-                        warn!("exception in tipper {:?}", e);
-                        match e {
-                            Error::ClientError(electrum_client::types::Error::JSON(_)) => {
-                                info!("tipper Client error, doing nothing")
-                            }
-                            _ => {
-                                warn!("trying to recreate died tipper client, {:?}", e);
-                                match &mut tipper {
-                                    TipperKind::Plain(tipper, url) => {
-                                        if let Ok(client) =
-                                            electrum_client::Client::new(url.as_str())
-                                        {
-                                            info!("succesfully created new tipper client");
-                                            tipper.client = client;
-                                        }
-                                    }
-                                    TipperKind::Tls(tipper, url, validate) => {
-                                        if let Ok(client) = electrum_client::Client::new_ssl(
-                                            url.as_str(),
-                                            *validate,
-                                        ) {
-                                            info!("succesfully created new tipper client");
-                                            tipper.client = client;
-                                        }
-                                    }
-                                }
+                if let Ok(client) = tipper_url.build_client() {
```

Comment:
Is there a particular reason you preferred creating a new socket every tick instead of reusing the same one?

Reply:
reusing the same would need to handle going offline or socket error, while reconnecting would handle that, and from the log of the http lib used it looks https session is reused, causing the penalty for reconnecting low (IIRC 10ms)

Reply:
Ok, yes, makes sense if the library reuses the connection.

Reply:
But wait, https? I guess you meant tls socket?

Reply:
yes, I meant tls socket